src/adk/api/local_api.py

```diff
@@ -1517,9 +1555,105 @@ def _validate_experiment_channels(self, experiment_file_path: Path, experiment_d
+                    expected_channel_info = self._get_channel_info(channel["slug"])
+
+                    for node in ["node1", "node2"]:
+                        if  expected_channel_info and channel[node] != expected_channel_info[node]:
+                            error_dict["error"].append(
+                                f"In file '{experiment_file_path}': Value '{channel[node]}' of Node '{node}' in channel"
+                                f" '{channel['slug']}' does not exist or is not a valid node for the channel")
+                    # check parameters in channel are valid and their value is within max and min range
+                    self._validate_template_parameters(entity='channel', entity_slug=channel["slug"],
+                                                       entity_params=channel["parameters"],
+                                                       experiment_file_path=experiment_file_path,
+                                                       error_dict=error_dict)
         else:
             error_dict["error"].append(f"No channels found for network '{experiment_network_slug}'")
 
+    def _validate_template_parameters(self, entity: str, entity_slug: str, entity_params: List[Dict[str, Any]],
+                                      experiment_file_path: Path, error_dict: ErrorDictType) -> None:
+        """
+        Validate the template parameters by checking:
+         - if the type of value is valid (integer or float)
+         - the value is within the range specified by the max and min value for the parameter
+
+        Args:
+            entity: The entity to which parameters belong to
+            entity_slug: The slug to identify the entity
+            entity_params: A dictionary containing the parameters to be validated
+            experiment_file_path: The location of the experiment.json file
+            error_dict: Dictionary containing error and warning messages of the validations that failed
+        """
+        param_templates = self._get_templates()
+        param_max_min_dict = self._get_template_params_max_min_range()
+
+        for param in entity_params: # pylint: disable=R1702 too-many-nested-blocks
+            if param["slug"] in param_templates:
+                for value in param["values"]:
+                    if value["name"] in param_max_min_dict[param["slug"]]:
+                        min_val = param_max_min_dict[param["slug"]][value['name']]['minimum_value']
+                        max_val = param_max_min_dict[param["slug"]][value['name']]['maximum_value']
+                        val = value["value"]
+                        if isinstance(val, (int, float)):
+                            if val < min_val or val > max_val:
+                                error_dict["error"].append(
+                                    f"In file '{experiment_file_path}': Value '{val}' of param"
+                                    f" '{param['slug']}' -> '{value['name']}' in {entity} '{entity_slug}'"
```

Comment:
Since the key value 'slug' is inside f-string { } , I am not able to use  \ (escape character) for escaping double quotes. 
I think using the single quotes is the best option here.  Same issue with the other three review items below.